### ISSUE TITLE
Injected arguments that should be null or undefined end up as empty object

### ISSUE THREAD

**Issue by zotanmew** [iyobo/amala#48]
Example:
```
@Controller('/test')
export class TestController {
	@Get('/whoami')
	@Flow(AuthMiddleware(false))
	async getTest(@CurrentUser() user: any): Promise<any> {
		console.log(user); // logs "{}" If ctx.state.user is null/undefined
		return user ?? "unknown"; // returns "{}" instead of "unknown" if ctx.state.user is null/undefined
	}
}
```

**Comment by zotanmew** [comment]
Traced to https://github.com/iyobo/amala/blob/master/src/util/generateRoutes.ts#L105, is null/undefined before and {} after, investigating more

**Comment by zotanmew** [comment]
I think this commit is responsible for the change in behavior https://github.com/iyobo/amala/commit/f98163d60a0ca223af285a27c789f93125ac9d8b#diff-ebf4f58c020ad0164c417b6a1a59310b7eb4c6662b80c0797793314434ff5c01L104-L106

**Comment by zotanmew** [comment]
Is this intended behavior? 🤔

**Comment by iyobo** [comment]
Definitely a bug. Assigning myself to fix this.

**Comment by iyobo** [comment]
Fixed in v10.0.4 and deployed to NPM.
